### ISSUE TITLE
mic.record() is incomplete

### ISSUE THREAD

**Issue by xaviliz** [respeaker/respeaker_python_library#47]
Hi all,

first of all, thanks for your dedication.

I have tried to use mic.record() but I have realize record function is incomplete or at least I cannot see wave.writeframes(data). Can you confirm it?

Thanks a lot,
XL

**Comment by xiongyihui** [comment]
You are right. The `record()` is not implemented. I will try to get it work

**Comment by xiongyihui** [comment]
Just add some code to implement the function. It should work. See https://github.com/respeaker/respeaker_python_library/blob/master/respeaker/microphone.py#L340

**Comment by xaviliz** [comment]
@xiongyihui thanks for your reply. To start with some test I have implemented an offline recorder. If I fix record() function for real-time, I will try to commit it. However I have tried to run some example with respeakerd but I have some issue with PulseAudio, I will open a new issue for that.
